demo/src/Components/BottomSheet/BottomSheet.tsx

```diff
@@ -17,74 +40,154 @@ const BottomSheet = (props: HvComponentProps) => {
+    stopPoints: props.element.getElementsByTagNameNS(
+      namespace,
+      BottomSheetStopPoint.localName,
+    ),
+    swipeToClose:
+      props.element.getAttributeNS(namespace, 'swipe-to-close') === 'true',
     toggleEventName: props.element.getAttributeNS(
       namespace,
       'toggle-event-name',
     ),
     visible: props.element.getAttributeNS(namespace, 'visible') === 'true',
   };
+  const hvStyles: HvStyles = {
+    bottomSheetBackgroundColor:
+      Hyperview.createStyleProp(props.element, props.stylesheets, {
+        ...props.options,
+        styleAttr: 'bottom-sheet:container-style',
+      })[0]?.backgroundColor || '#fff',
+    handle: Hyperview.createStyleProp(props.element, props.stylesheets, {
+      ...props.options,
+      styleAttr: 'bottom-sheet:handle-style',
+    }),
+    overlay: Hyperview.createStyleProp(props.element, props.stylesheets, {
+      ...props.options,
+      styleAttr: 'bottom-sheet:overlay-style',
+    }),
+  };
+
+  const [contentSectionHeights, setContentSectionHeights] = useState(
+    new Array(hvProps.contentSections.length),
+  );
+
+  const stopPointLocations = React.useMemo(() => {
+    return Array.from(hvProps.stopPoints)
+      .map<number | null>(element => {
+        const location = element.getAttributeNS(namespace, 'location');
+        return typeof element !== 'string' &&
+          element &&
+          location &&
+          typeof location === 'string'
+          ? parseFloat(location)
+          : null;
+      }, [])
+      .sort((a, b) => {
+        if (a !== null && b !== null) {
+          return a - b;
+        }
+        return 0;
+      });
+  }, [hvProps.stopPoints]);
 
   const [visible, setVisible] = useState(hvProps.visible);
   const [height, setHeight] = useState(0);
 
-  const translateY = useRef(new Animated.Value(0)).current;
-  const overlayOpacity = useRef(new Animated.Value(0)).current;
-  const contentOpacity = useRef(new Animated.Value(0)).current;
+  const overlayOpacity = useSharedValue(0);
+  const contentOpacity = useSharedValue(0);
+  const context = useSharedValue({ startTime: Date.now(), y: 0 });
+  const translateY = useSharedValue(0);
+  const [upcomingTranslateY, setUpcomingTranslateY] = useState(0);
+  const velocity = useSharedValue(0);
+  const targetOpacity: number = hvStyles.overlay[0]?.opacity ?? 1;
 
-  const onLayout = (event: LayoutChangeEvent) => {
-    setHeight(event.nativeEvent.layout.height);
+  const hide = () => {
+    setVisible(false);
   };
 
-  const targetOpacity: number = styles.overlay.opacity ?? 1;
+  const scrollTo = useCallback(
+    (destination: number) => {
+      'worklet';
+
+      // reanimated takes time to set translateY value but
+      // we need the new value to calculate whether innerView should scroll
+      // so we use upcomingTranslateY to handle this case
+      runOnJS(setUpcomingTranslateY)(destination);
+      translateY.value = withSpring(destination, { damping: 50 });
```

Comment:
nit: could we making the damping a const at the top of this file? Ideally in the future we might want to make it an XML attribute